### PR TITLE
controllervm controller

### PR DESCRIPTION
**Pull request by stuggi**
Creates controller VMs on CNV base on the controlplane CR. The
base image, network-attachment and virtual machine is created
as unstructured. A secret is created for the cloudinit userdata
using a template.
The controlplane controller create a secret for the deployment
user cloud-admin. The cloud-admin user and the pubkey from the
secret gets added via cloudinit.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,15 +27,40 @@ RUN go mod download
...
+        name="osp-director-operator" \
+        version="1.0" \
+        summary="OSP Director Operator" \
+        io.k8s.name="osp-director-operator" \
+        io.k8s.description="This image includes the osp-director-operator"
+
+# TODO: For now hard code the WATCH_NAMESPACE here
+ENV USER_UID=1001 \
+    OPERATOR_BINDATA_DIR=/bindata/ \
+    OPERATOR_TEMPLATES=/usr/share/nova-operator/templates/ \
```
Comment (dprince) [inline Dockerfile:52]: nova-operator?

File: go.mod [hunk 2]
```diff
@@ -3,9 +3,17 @@ module github.com/openstack-k8s-operators/osp-director-operator
 go 1.13
 
 require (
+	github.com/Masterminds/sprig v2.22.0+incompatible
 	github.com/go-logr/logr v0.1.0
 	github.com/onsi/ginkgo v1.12.1
 	github.com/onsi/gomega v1.10.1
+	github.com/openshift/api v3.9.0+incompatible
+	github.com/openstack-k8s-operators/lib-common v0.0.0-20201012132655-247b83b2fafa
+	github.com/openstack-k8s-operators/nova-operator v0.0.0-20201020141434-6a40ed962c36
```
Comment (dprince) [inline go.mod:12]: do we require lib-common and nova-operator?

File: Dockerfile (hunk 1 above)
Comment (stuggi) [reply Dockerfile]: wrong c/p

File: go.mod (hunk 2 above)
Comment (stuggi) [reply go.mod]: no we don't, thats got added in an initial version and missed to remove them

**Review by dprince — APPROVED** [review]

**Pull request merged**